waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -813,11 +819,13 @@
                 (try
                   (loop [version (reset-watch-state! scheduler options)
                          iter 0]
-                    (let [watch-url (str resource-url "&watch=true&resourceVersion=" version)]
-                      ;; process updates forever (unless there's an exception)
-                      (doseq [json-object (streaming-api-request-fn watch-url)]
-                        (when json-object
-                          (update-fn json-object))))
+                    (timers/start-stop-time!
```

Comment:
Looking at [the definition of `start-stop-time!`](https://github.com/metrics-clojure/metrics-clojure/blob/fe6b5f44e2a661cfbdd3dd09ee43567c030b5ff3/metrics-clojure-core/src/metrics/timers.clj#L139-L144), it doesn't appear to handle exceptions gracefully. I don't think that's an issue here, but we should keep that in mind.